src/families/stellar/bridge/mock.js

```diff
@@ -0,0 +1,270 @@
+  }
+
+  if (!t.fees || !t.baseReserve) {
+    errors.fees = new FeeNotLoaded();
+  }
+
+  let estimatedFees = !t.fees ? BigNumber(0) : t.fees;
+  let baseReserve = !t.baseReserve ? BigNumber(0) : t.baseReserve;
+
+  let amount = !useAllAmount
+    ? t.amount
+    : a.balance.minus(baseReserve).minus(estimatedFees);
+  let totalSpent = !useAllAmount
+    ? amount.plus(estimatedFees)
+    : a.balance.minus(baseReserve);
+
+  if (useAllAmount) {
+    warnings.amount = new NotEnoughSpendableBalance("", {
+      minimumBalance: `${baseReserve.toString()} XLM`
+    });
+  }
+
+  if (
+    !errors.amount &&
+    amount
+      .plus(estimatedFees)
+      .plus(baseReserve)
+      .gt(a.balance)
+  ) {
+    errors.amount = new NotEnoughBalance();
+  }
+
+  if (
+    !errors.recipient &&
+    !errors.amount &&
+    (amount.lt(0) || totalSpent.gt(a.balance))
+  ) {
+    errors.amount = new NotEnoughBalance();
+    totalSpent = BigNumber(0);
+    amount = BigNumber(0);
+  }
+
+  if (!errors.amount && amount.eq(0)) {
+    errors.amount = new AmountRequired();
+  }
+
+  // if amount < 1.0 you can't
+  if (
+    !errors.amount &&
+    notCreatedAddresses.includes(t.recipient) &&
+    amount.lt(10000000)
+  ) {
+    errors.amount = new NotEnoughBalanceBecauseDestinationNotCreated("", {
+      minimalAmount: "1 XLM"
+    });
+  }
+
+  if (t.memoType && t.memoValue && !isMemoValid(t.memoType, t.memoValue)) {
+    errors.transaction = new StellarWrongMemoFormat();
+  }
+
+  return Promise.resolve({
+    errors,
+    warnings,
+    estimatedFees,
+    amount,
+    totalSpent
+  });
+};
+
+const prepareTransaction = async (a, t) => {
+  const networkInfo = t.networkInfo || {
+    family: "stellar",
+    fees: BigNumber("100"),
+    baseReserve: BigNumber("100000")
+  };
+  invariant(networkInfo.family === "stellar", "stellar networkInfo expected");
+
+  const fees = t.fees || networkInfo.fees;
+  const baseReserve = t.baseReserve || networkInfo.baseReserve;
+
+  const getMemoData = async () => {
+    if (t.memoType) {
+      return {
+        memoType: t.memoType,
+        memoTypeRecommended: t.memoTypeRecommended
+      };
+    } else {
+      if (!isInvalidRecipient(t.recipient)) {
+        const memoType = memotypeTextRecommendedAddresses.includes(t.recipient)
+          ? "MEMO_TEXT"
+          : null;
+        const memoTypeRecommended = memoType !== null ? true : false;
+        return { memoType, memoTypeRecommended };
+      }
+      return {
+        memoType: undefined,
+        memoTypeRecommended: t.memoTypeRecommended
+      };
+    }
+  };
+
+  const { memoType, memoTypeRecommended } = await getMemoData();
+
+  if (
+    t.networkInfo !== networkInfo ||
+    t.fees !== fees ||
+    t.baseReserve !== baseReserve ||
+    t.memoType !== memoType
+  ) {
+    return {
+      ...t,
+      networkInfo,
+      fees,
+      baseReserve,
+      memoType,
+      memoTypeRecommended
+    };
+  }
+
+  return t;
+};
+
+const estimateMaxSpendable = async ({
```

Comment:
Would this not benefit from what you did @gre for swap

Reply:
I don't understand the question.
yes it implements that method 🤔